### PR TITLE
Fix resetting PohRecorder to wrong bank

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Resetting the PohRecorder to the wrong bank on skipping slots causes future PohRecorder tick height to be out of sync with bank tick heights for future leader slots.

#### Summary of Changes
Reset PohRecorder to correct bank.

Fixes #

### CODE REVIEW

**Comment by garious** [comment]
Any thoughts on what the code that demonstrates the failure should look like?

**Comment by pgarg66** [comment]
@carllin , do you want to cherry pick this to v12? We can include it in .4 tag

**Comment by carllin** [comment]
@pgarg66 there's a better fix coming, will cherry-pick after landing that one

File: core/src/leader_schedule_utils.rs [hunk 1]
```diff
@@ -59,6 +65,12 @@ pub fn next_leader_slot(pubkey: &Pubkey, mut current_slot: u64, bank: &Bank) ->
         for i in start_index..bank.get_slots_in_epoch(epoch) {
             current_slot += 1;
             if *pubkey == leader_schedule[i] {
+                if let Some(blocktree) = blocktree {
```
Comment (carllin) [inline core/src/leader_schedule_utils.rs]: this is the only non-testing changes xD

**Pull request merged**